File: Data Structure/Heap/LET_no1086_High Five.py
class Solution:
    def highFive(self, items: List[List[int]]) -> List[List[int]]:
        memo = collections.defaultdict(list)
        for ID, score in items:
            heapq.heappush(memo[ID], score)
            if len(memo[ID]) > 5:
                heapq.heappop(memo[ID])
        
        return [[ID, sum(memo[ID]) // 5] for ID in sorted(memo)]

File: Data Structure/Heap/LET_no1086_High Five.cpp
class Solution {
public:
    vector<vector<int>> highFive(vector<vector<int>>& items) {
        unordered_map<int, priority_queue<int, vector<int>, greater<int>>> memo;
        for (vector<int>& item : items) {
            memo[item[0]].push(item[1]);
            if (memo[item[0]].size() > 5)
                memo[item[0]].pop();
        }

        vector<vector<int>> res;
        for (auto& [key, vals] : memo) {
            int cur = 0;
            while (!vals.empty()) {
                cur += vals.top();
                vals.pop();
            }
            res.push_back({key, cur / 5});
        }

        sort(res.begin(), res.end(), [](vector<int>& a, vector<int>& b){ return a[0] < b[0]; });
        return res;
    }
};